QPlatformWindow::setWindowState API changed in Qt5.11

Arch GNU/Linux, `bionic` branch
API Change: https://github.com/qt/qtbase/commit/a02959bb5b43a3f9d881e5213ceedf535202b6a1#diff-485e8544fe6332d7d6cf505525a03335
```
In file included from qmirclientclipboard.cpp:43:
qmirclientwindow.h:76:10: error: ‘void QMirClientWindow::setWindowState(Qt::WindowState)’ marked ‘override’, but does not override
     void setWindowState(Qt::WindowState state) override;
          ^~~~~~~~~~~~~~
```

--- a/src/ubuntumirclient/qmirclientwindow.cpp
+++ b/src/ubuntumirclient/qmirclientwindow.cpp
@@ -56,6 +56,7 @@
 #include <QSize>
 #include <QtMath>
 #include <QtGui/private/qguiapplication_p.h>
+#include <QtGui/private/qwindow_p.h>
 #include <QtEglSupport/private/qeglconvenience_p.h>
 
 #include <EGL/egl.h>
@@ -853,13 +854,13 @@ void QMirClientWindow::handleSurfaceStateChanged(Qt::WindowState state)
     QWindowSystemInterface::handleWindowStateChanged(window(), state);
 }
 
-void QMirClientWindow::setWindowState(Qt::WindowState state)
+void QMirClientWindow::setWindowState(Qt::WindowStates state)
 {
     QMutexLocker lock(&mMutex);
-    qCDebug(mirclient, "setWindowState(window=%p, %s)", this, qtWindowStateToStr(state));
+    qCDebug(mirclient, "setWindowState(window=%p, %s)", this, qtWindowStateToStr(QWindowPrivate::effectiveState(state)));
 
-    if (mWindowState == state) return;
-    mWindowState = state;
+    if (mWindowState == QWindowPrivate::effectiveState(state)) return;
+    mWindowState = QWindowPrivate::effectiveState(state);
 
     lock.unlock();
     updateSurfaceState();

--- a/src/ubuntumirclient/qmirclientwindow.h
+++ b/src/ubuntumirclient/qmirclientwindow.h
@@ -73,7 +73,7 @@ public:
     WId winId() const override;
     QRect geometry() const override;
     void setGeometry(const QRect&) override;
-    void setWindowState(Qt::WindowState state) override;
+    void setWindowState(Qt::WindowStates state) override;
     void setWindowFlags(Qt::WindowFlags flags) override;
     void setVisible(bool visible) override;
     void setWindowTitle(const QString &title) override;

